modules/core/src/system.cpp

```diff
@@ -1300,6 +1300,402 @@ void setUseCollection(bool flag)
+        {
+            if(queue[i] == command)
+            {
+                processCommand(command);
+                queue.erase(queue.begin()+i);
+            }
+        }
+    }
+#else
+    CV_UNUSED(command);
+#endif
+}
+
+void setFlags(int modeFlags)
+{
+#ifdef ENABLE_INSTRUMENTATION
+    getInstrumentStruct().enableMapping = (modeFlags & FLAGS_MAPPING);
+#else
+    CV_UNUSED(modeFlags);
+#endif
+}
+int getFlags()
+{
+#ifdef ENABLE_INSTRUMENTATION
+    int flags = 0;
+    if(getInstrumentStruct().enableMapping)
+        flags |= FLAGS_MAPPING;
+    return flags;
+#else
+    return 0;
+#endif
+}
+
+NodeData::NodeData(const char* funName, int instrType, int implType)
+{
+    m_instrType = TYPE_GENERAL;
+    m_implType  = IMPL_PLAIN;
+
+    m_funError  = false;
+    m_counter   = 0;
+    m_ticksAcc  = 0;
+    m_stopPoint = false;
+    m_mtRegion  = false;
+    m_pRefNode  = NULL;
+    m_pTls      = NULL;
+
+    m_funName     = funName;
+    m_instrType   = instrType;
+    m_implType    = implType;
+}
+NodeData::NodeData(NodeData &ref)
+{
+    *this = ref;
+}
+NodeData& NodeData::operator=(const NodeData &right)
+{
+    this->m_funName     = right.m_funName;
+    this->m_instrType   = right.m_instrType;
+    this->m_implType    = right.m_implType;
+    this->m_funError    = right.m_funError;
+    this->m_counter     = right.m_counter;
+    this->m_ticksAcc    = right.m_ticksAcc;
+    this->m_stopPoint   = right.m_stopPoint;
+    this->m_mtRegion    = right.m_mtRegion;
+    this->m_pTls        = NULL;
+
+    return *this;
+}
+NodeData::~NodeData()
+{
+    if(m_pTls)
+    {
+        delete m_pTls;
+        m_pTls = NULL;
+    }
+}
+bool operator==(const NodeData& left, const NodeData& right)
+{
+    if(left.m_funName == right.m_funName)
+        return true;
+    return false;
+}
+
+#ifdef ENABLE_INSTRUMENTATION
+InstrStruct& getInstrumentStruct()
+{
+    static InstrStruct instr;
+    return instr;
+}
+
+InstrTLSStruct& getInstrumentTLSStruct()
+{
+    return *getInstrumentStruct().tlsStruct.get();
+}
+
+InstrNode* getCurrentNode()
+{
+    return getInstrumentTLSStruct().pCurrentNode;
+}
+
+IntrumentationRegion::IntrumentationRegion(const char* funName, const char*, int, int instrType, int implType, bool mt)
+{
+    m_disabled    = false;
+    m_regionTicks = 0;
+    m_mtPoint     = false;
+
+    InstrStruct *pStruct = &getInstrumentStruct();
+    if(pStruct->useInstr)
+    {
+        m_mtPoint = mt;
+        InstrTLSStruct *pTLS = &getInstrumentTLSStruct();
+        if(!pTLS->pCurrentNode)
+        {
+            pTLS->pCurrentNode = pStruct->pRefNode; // Initialize TLS node for thread
+        }
+
+        // Disable in case of failure
+        if(!pTLS->pCurrentNode)
+        {
+            CV_DbgAssert(pTLS->pCurrentNode != NULL);
+            m_disabled = true;
+            return;
+        }
+
+        m_disabled = pTLS->pCurrentNode->m_payload.m_stopPoint;
+        if(m_disabled)
+            return;
```

Comment:
Stop point is a signal to stop looking for childs in case we stopped mapping trace or there was an error. m_disabled is the same thing just for the region object. In this case tracer should continue to work with existing calls map.